### ISSUE TITLE
EC2 instance type

### ISSUE THREAD

**Issue by JoshHilliker** [aws-observability/aws-otel-collector#69]
Is there a minimum EC2 instance the package will run on?  I'm using a t2-micro to test out and the RPM instructions are failing.

**Comment by mxiamxia** [comment]
Hi @JoshHilliker, There is no requirement on EC2 instance type. The collector can be installed and ran on `t2.micro`. Can you please provide the following info and we can help you on the issue?

1, what' was the EC2 ami_id you were trying to test on?
2, Did you test it on x86 or arm EC2 host?
3, Can you please provide the error details or what kind of failure you had seen?

Please let us know any question you have. Ty!

**Comment by JoshHilliker** [comment]
1.  ami-0947d2ba12ee1ff75
2. x86 - Intel(R) Xeon(R) CPU E5-2676 v3 @ 2.40GHz
3. 

sudo make package-rpm
ARCH=x86_64 DEST=build/packages/linux/amd64 tools/packaging/linux/create_rpm.sh
**********************************************************
Creating rpm file for Amazon Linux and RHEL, Arch: x86_64
**********************************************************
Creating rpmbuild workspace
Creating file structure
Copying application files
cp: cannot stat ‘build/linux/aoc_linux_x86_64’: No such file or directory
make: *** [package-rpm] Error 1

**Comment by mxiamxia** [comment]
There is one step missed. Can you please run `sudo make build` before `sudo make package-rpm`? I'll update the instruction if it works for you. Thanks.

**Comment by JoshHilliker** [comment]
sudo make build - error below. 
staticcheck FAILED => static check errors:\n
/bin/sh: staticcheck: command not found\n
make: *** [lint-static-check] Error 1

**Comment by JoshHilliker** [comment]
aws-otel-collector]$ sudo make build
go install github.com/golangci/golangci-lint/cmd/golangci-lint
go: downloading github.com/golangci/golangci-lint v1.31.0
go: extracting github.com/golangci/golangci-lint v1.31.0
go: downloading golang.org/x/tools v0.0.0-20200822203824-307de81be3f4
go: downloading github.com/gofrs/flock v0.8.0
go: downloading github.com/go-xmlfmt/xmlfmt v0.0.0-20191208150333-d5b6f63a941b
go: downloading github.com/tommy-muehle/go-mnd v1.3.1-0.20200224220436-e6f9a994e                               8fa
go: downloading github.com/nakabonne/nestif v0.3.0
go: downloading honnef.co/go/tools v0.0.1-2020.1.5
go: extracting github.com/gofrs/flock v0.8.0
go: downloading github.com/ultraware/funlen v0.0.3
go: extracting github.com/tommy-muehle/go-mnd v1.3.1-0.20200224220436-e6f9a994e8                               fa
go: downloading github.com/golangci/revgrep v0.0.0-20180526074752-d9c87f5ffaf0
go: extracting github.com/go-xmlfmt/xmlfmt v0.0.0-20191208150333-d5b6f63a941b
go: extracting github.com/nakabonne/nestif v0.3.0
go: downloading github.com/jirfag/go-printf-func-name v0.0.0-20191110105641-45db                               9963cdd3
go: extracting github.com/golangci/revgrep v0.0.0-20180526074752-d9c87f5ffaf0
go: downloading github.com/shazow/go-diff v0.0.0-20160112020656-b6b7b6733b8c
go: extracting github.com/ultraware/funlen v0.0.3
go: extracting github.com/shazow/go-diff v0.0.0-20160112020656-b6b7b6733b8c
go: downloading github.com/ultraware/whitespace v0.0.4
go: extracting github.com/jirfag/go-printf-func-name v0.0.0-20191110105641-45db9                               963cdd3
go: extracting honnef.co/go/tools v0.0.1-2020.1.5
go: extracting github.com/ultraware/whitespace v0.0.4
go: downloading github.com/golangci/errcheck v0.0.0-20181223084120-ef45e06d44b6
go: extracting github.com/golangci/errcheck v0.0.0-20181223084120-ef45e06d44b6
go: extracting golang.org/x/tools v0.0.0-20200822203824-307de81be3f4
go: downloading mvdan.cc/interfacer v0.0.0-20180901003855-c20040233aed
go: downloading github.com/golangci/ineffassign v0.0.0-20190609212857-42439a7714cc
go: downloading github.com/maratori/testpackage v1.0.1
go: downloading mvdan.cc/gofumpt v0.0.0-20200709182408-4fd085cb6d5f
go: downloading github.com/kyoh86/exportloopref v0.1.7
go: downloading github.com/denis-tingajkin/go-header v0.3.1
go: downloading github.com/bombsimon/wsl/v3 v3.1.0
go: downloading github.com/nishanths/exhaustive v0.0.0-20200811152831-6cf413ae40e0
go: downloading github.com/securego/gosec v0.0.0-20200203094520-d13bb6d2420c
go: downloading github.com/uudashr/gocognit v1.0.1
go: extracting github.com/maratori/testpackage v1.0.1
go: downloading github.com/golangci/goconst v0.0.0-20180610141641-041c5f2b40f3
go: extracting mvdan.cc/interfacer v0.0.0-20180901003855-c20040233aed
go: downloading github.com/sourcegraph/go-diff v0.6.0
go: extracting github.com/golangci/ineffassign v0.0.0-20190609212857-42439a7714cc
go: downloading github.com/tdakkota/asciicheck v0.0.0-20200416190851-d7f85be797a2
go: extracting github.com/sourcegraph/go-diff v0.6.0
go: extracting github.com/kyoh86/exportloopref v0.1.7
go: downloading github.com/Djarvur/go-err113 v0.0.0-20200511133814-5174e21577d5
go: extracting github.com/nishanths/exhaustive v0.0.0-20200811152831-6cf413ae40e0
go: downloading mvdan.cc/unparam v0.0.0-20190720180237-d51796306d8f
go: extracting github.com/bombsimon/wsl/v3 v3.1.0
go: downloading github.com/daixiang0/gci v0.2.4
go: extracting github.com/denis-tingajkin/go-header v0.3.1
go: downloading github.com/kisielk/gotool v1.0.0
go: extracting github.com/securego/gosec v0.0.0-20200203094520-d13bb6d2420c
go: downloading github.com/securego/gosec/v2 v2.4.0
go: extracting github.com/uudashr/gocognit v1.0.1
go: downloading github.com/golangci/check v0.0.0-20180506172741-cfe4005ccda2
go: downloading github.com/sonatard/noctx v0.0.1
go: extracting mvdan.cc/gofumpt v0.0.0-20200709182408-4fd085cb6d5f
go: extracting github.com/Djarvur/go-err113 v0.0.0-20200511133814-5174e21577d5
go: downloading golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
go: extracting mvdan.cc/unparam v0.0.0-20190720180237-d51796306d8f
go: downloading github.com/golangci/gofmt v0.0.0-20190930125516-244bba706f1a
go: extracting github.com/daixiang0/gci v0.2.4
go: downloading github.com/golangci/misspell v0.0.0-20180809174111-950f5d19e770
go: extracting github.com/golangci/gofmt v0.0.0-20190930125516-244bba706f1a
go: extracting github.com/golangci/goconst v0.0.0-20180610141641-041c5f2b40f3
go: downloading github.com/timakin/bodyclose v0.0.0-20190930140734-f7f2e9bca95e
go: downloading github.com/golangci/dupl v0.0.0-20180902072040-3e9179ac440a
go: extracting github.com/tdakkota/asciicheck v0.0.0-20200416190851-d7f85be797a2
go: downloading github.com/go-critic/go-critic v0.5.2
go: extracting github.com/golangci/check v0.0.0-20180506172741-cfe4005ccda2
go: downloading github.com/ryancurrah/gomodguard v1.1.0
go: downloading github.com/golangci/unconvert v0.0.0-20180507085042-28b1c447d1f4
go: extracting github.com/timakin/bodyclose v0.0.0-20190930140734-f7f2e9bca95e
go: downloading github.com/jingyugao/rowserrcheck v0.0.0-20191204022205-72ab7603b68a
go: extracting github.com/kisielk/gotool v1.0.0
go: downloading github.com/gostaticanalysis/analysisutil v0.0.3
go: extracting github.com/securego/gosec/v2 v2.4.0
go: downloading golang.org/x/mod v0.3.0
go: extracting github.com/golangci/dupl v0.0.0-20180902072040-3e9179ac440a
go: downloading github.com/golangci/gocyclo v0.0.0-20180528144436-0a533e8fa43d
go: extracting golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
go: downloading github.com/ryanrolds/sqlclosecheck v0.3.0
go: extracting github.com/sonatard/noctx v0.0.1
go: downloading github.com/nbutton23/zxcvbn-go v0.0.0-20180912185939-ae427f1e4c1d
go: extracting github.com/golangci/misspell v0.0.0-20180809174111-950f5d19e770
go: extracting github.com/go-critic/go-critic v0.5.2
go: extracting github.com/ryancurrah/gomodguard v1.1.0
go: downloading github.com/phayes/checkstyle v0.0.0-20170904204023-bfd46e6a821d
go: extracting github.com/golangci/unconvert v0.0.0-20180507085042-28b1c447d1f4
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: extracting github.com/jingyugao/rowserrcheck v0.0.0-20191204022205-72ab7603b68a
go: downloading github.com/golangci/maligned v0.0.0-20180506175553-b1d89398deca
go: extracting github.com/gostaticanalysis/analysisutil v0.0.3
go: downloading github.com/OpenPeeDeeP/depguard v1.0.1
go: extracting golang.org/x/mod v0.3.0
go: extracting github.com/golangci/gocyclo v0.0.0-20180528144436-0a533e8fa43d
go: downloading github.com/golangci/lint-1 v0.0.0-20191013205115-297bf364a8e0
go: extracting github.com/ryanrolds/sqlclosecheck v0.3.0
go: downloading mvdan.cc/lint v0.0.0-20170908181259-adc824a0674b
go: downloading github.com/Masterminds/semver v1.5.0
go: extracting github.com/Masterminds/semver v1.5.0
go: extracting github.com/phayes/checkstyle v0.0.0-20170904204023-bfd46e6a821d
go: downloading github.com/BurntSushi/toml v0.3.1
go: extracting github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/golangci/go-misc v0.0.0-20180628070357-927a3d87b613
go: extracting github.com/golangci/maligned v0.0.0-20180506175553-b1d89398deca
go: downloading github.com/golangci/prealloc v0.0.0-20180630174525-215b22d4de21
go: extracting github.com/OpenPeeDeeP/depguard v1.0.1
go: downloading github.com/ssgreg/nlreturn/v2 v2.1.0
go: extracting github.com/golangci/lint-1 v0.0.0-20191013205115-297bf364a8e0
go: downloading github.com/tetafro/godot v0.4.8
go: extracting mvdan.cc/lint v0.0.0-20170908181259-adc824a0674b
go: downloading github.com/gobwas/glob v0.2.3
go: downloading github.com/go-toolsmith/astfmt v1.0.0
go: downloading github.com/go-toolsmith/astequal v1.0.0
go: downloading github.com/go-toolsmith/strparse v1.0.0
go: extracting github.com/go-toolsmith/astequal v1.0.0
go: downloading github.com/go-toolsmith/astcopy v1.0.0
go: extracting github.com/BurntSushi/toml v0.3.1
go: extracting github.com/golangci/go-misc v0.0.0-20180628070357-927a3d87b613
go: downloading github.com/quasilyte/go-ruleguard v0.2.0
go: extracting github.com/tetafro/godot v0.4.8
go: downloading github.com/quasilyte/regex/syntax v0.0.0-20200407221936-30656e2c4a95
go: downloading github.com/matoous/godox v0.0.0-20190911065817-5d6d842e92eb
go: extracting github.com/nbutton23/zxcvbn-go v0.0.0-20180912185939-ae427f1e4c1d
go: extracting github.com/golangci/prealloc v0.0.0-20180630174525-215b22d4de21
go: downloading github.com/go-toolsmith/astcast v1.0.0
go: extracting github.com/go-toolsmith/astfmt v1.0.0
go: downloading github.com/go-toolsmith/astp v1.0.0
go: extracting github.com/ssgreg/nlreturn/v2 v2.1.0
go: downloading github.com/stretchr/objx v0.2.0
go: extracting github.com/go-toolsmith/strparse v1.0.0
go: downloading github.com/go-toolsmith/typep v1.0.2
go: extracting github.com/go-toolsmith/astcopy v1.0.0
go: extracting github.com/quasilyte/go-ruleguard v0.2.0
go: extracting github.com/matoous/godox v0.0.0-20190911065817-5d6d842e92eb
go: extracting github.com/go-toolsmith/astcast v1.0.0
go: extracting github.com/go-toolsmith/astp v1.0.0
go: extracting github.com/stretchr/objx v0.2.0
go: extracting github.com/quasilyte/regex/syntax v0.0.0-20200407221936-30656e2c4a95
go: extracting github.com/go-toolsmith/typep v1.0.2
go: extracting github.com/gobwas/glob v0.2.3
go: finding github.com/golangci/golangci-lint v1.31.0
go: finding github.com/gofrs/flock v0.8.0
go: finding github.com/go-critic/go-critic v0.5.2
go: finding github.com/stretchr/objx v0.2.0
go: finding golang.org/x/tools v0.0.0-20200822203824-307de81be3f4
go: finding github.com/go-toolsmith/astfmt v1.0.0
go: finding github.com/mitchellh/go-homedir v1.1.0
go: finding golang.org/x/mod v0.3.0
go: finding github.com/go-toolsmith/astequal v1.0.0
go: finding golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
go: finding github.com/go-toolsmith/astp v1.0.0
go: finding github.com/go-toolsmith/typep v1.0.2
go: finding github.com/go-toolsmith/astcast v1.0.0
go: finding github.com/go-toolsmith/astcopy v1.0.0
go: finding github.com/Djarvur/go-err113 v0.0.0-20200511133814-5174e21577d5
go: finding github.com/go-toolsmith/strparse v1.0.0
go: finding github.com/quasilyte/go-ruleguard v0.2.0
go: finding github.com/OpenPeeDeeP/depguard v1.0.1
go: finding github.com/quasilyte/regex/syntax v0.0.0-20200407221936-30656e2c4a95
go: finding github.com/go-xmlfmt/xmlfmt v0.0.0-20191208150333-d5b6f63a941b
go: finding github.com/bombsimon/wsl/v3 v3.1.0
go: finding github.com/gobwas/glob v0.2.3
go: finding github.com/daixiang0/gci v0.2.4
go: finding github.com/denis-tingajkin/go-header v0.3.1
go: finding github.com/golangci/check v0.0.0-20180506172741-cfe4005ccda2
go: finding github.com/golangci/dupl v0.0.0-20180902072040-3e9179ac440a
go: finding github.com/golangci/errcheck v0.0.0-20181223084120-ef45e06d44b6
go: finding github.com/golangci/go-misc v0.0.0-20180628070357-927a3d87b613
go: finding github.com/golangci/goconst v0.0.0-20180610141641-041c5f2b40f3
go: finding github.com/golangci/revgrep v0.0.0-20180526074752-d9c87f5ffaf0
go: finding github.com/golangci/gocyclo v0.0.0-20180528144436-0a533e8fa43d
go: finding github.com/golangci/gofmt v0.0.0-20190930125516-244bba706f1a
go: finding github.com/golangci/ineffassign v0.0.0-20190609212857-42439a7714cc
go: finding github.com/golangci/lint-1 v0.0.0-20191013205115-297bf364a8e0
go: finding github.com/golangci/maligned v0.0.0-20180506175553-b1d89398deca
go: finding github.com/golangci/misspell v0.0.0-20180809174111-950f5d19e770
go: finding github.com/golangci/prealloc v0.0.0-20180630174525-215b22d4de21
go: finding github.com/golangci/unconvert v0.0.0-20180507085042-28b1c447d1f4
go: finding github.com/kisielk/gotool v1.0.0
go: finding github.com/jingyugao/rowserrcheck v0.0.0-20191204022205-72ab7603b68a
go: finding github.com/jirfag/go-printf-func-name v0.0.0-20191110105641-45db9963cdd3
go: finding github.com/gostaticanalysis/analysisutil v0.0.3
go: finding github.com/kyoh86/exportloopref v0.1.7
go: finding github.com/maratori/testpackage v1.0.1
go: finding github.com/matoous/godox v0.0.0-20190911065817-5d6d842e92eb
go: finding github.com/nakabonne/nestif v0.3.0
go: finding github.com/nishanths/exhaustive v0.0.0-20200811152831-6cf413ae40e0
go: finding github.com/ryancurrah/gomodguard v1.1.0
go: finding github.com/ryanrolds/sqlclosecheck v0.3.0
go: finding github.com/Masterminds/semver v1.5.0
go: finding github.com/securego/gosec/v2 v2.4.0
go: finding github.com/phayes/checkstyle v0.0.0-20170904204023-bfd46e6a821d
go: finding github.com/shazow/go-diff v0.0.0-20160112020656-b6b7b6733b8c
go: finding github.com/nbutton23/zxcvbn-go v0.0.0-20180912185939-ae427f1e4c1d
go: finding github.com/sonatard/noctx v0.0.1
go: finding github.com/sourcegraph/go-diff v0.6.0
go: finding github.com/ssgreg/nlreturn/v2 v2.1.0
go: finding github.com/tdakkota/asciicheck v0.0.0-20200416190851-d7f85be797a2
go: finding github.com/tetafro/godot v0.4.8
go: finding github.com/timakin/bodyclose v0.0.0-20190930140734-f7f2e9bca95e
go: finding github.com/tommy-muehle/go-mnd v1.3.1-0.20200224220436-e6f9a994e8fa
go: finding github.com/ultraware/funlen v0.0.3
go: finding github.com/ultraware/whitespace v0.0.4
go: finding github.com/uudashr/gocognit v1.0.1
go: finding honnef.co/go/tools v0.0.1-2020.1.5
go: finding mvdan.cc/gofumpt v0.0.0-20200709182408-4fd085cb6d5f
go: finding github.com/BurntSushi/toml v0.3.1
go: finding mvdan.cc/interfacer v0.0.0-20180901003855-c20040233aed
go: finding mvdan.cc/unparam v0.0.0-20190720180237-d51796306d8f
go: finding mvdan.cc/lint v0.0.0-20170908181259-adc824a0674b
go install github.com/jstemmer/go-junit-report
go: downloading github.com/jstemmer/go-junit-report v0.9.1
go: extracting github.com/jstemmer/go-junit-report v0.9.1
go: finding github.com/jstemmer/go-junit-report v0.9.1
go install honnef.co/go/tools/cmd/staticcheck
staticcheck FAILED => static check errors:\n
/bin/sh: staticcheck: command not found\n
make: *** [lint-static-check] Error 1

**Comment by wyTrivail** [comment]
i reckon your golang home is not in your system PATH, could you check? meanwhile, i will try to fix this issue

**Comment by JoshHilliker** [comment]
I checked. .  $GOPATH is setup correctly.

**Comment by MahekPavagadhi** [comment]
I had the same issue yesterday with the `sudo make build `. I executed the following steps and was able to successfully install collector on Linux EC2 instance.

`unset GOROOT`
 `export GOPATH=`where you want to install staticcheck binary

I set here 
```
echo $GOPATH/
/home/ec2-user/go/
```

Install staticcheck 

`go get honnef.co/go/tools/cmd/staticcheck `

Validate the installation - It will go inside bin of your GOPATH:
```
which staticcheck 
/home/ec2-user/go/bin/staticcheck
```

I made following changes in Makefile

```
LINT=/home/ec2-user/go/bin/golangci-lint
STATIC_CHECK=/home/ec2-user/go/bin/staticcheck
```

<img width="1285" alt="image" src="https://user-images.githubusercontent.com/6774570/97476963-4736ae80-190c-11eb-87e6-3b593b67f0f8.png">

After this I executed `sudo make package-rpm` and `sudo rpm -Uvh  ./aws-otel-collector.rpm` to successfully install collector

<img width="636" alt="image" src="https://user-images.githubusercontent.com/6774570/97476048-35a0d700-190b-11eb-822d-ea2ea3fbebf8.png">


<img width="1195" alt="image" src="https://user-images.githubusercontent.com/6774570/97476390-93352380-190b-11eb-8b7f-307e380de2bd.png">

**Comment by mxiamxia** [comment]
@MahekPavagadhi , @JoshHilliker . I just send a PR https://github.com/aws-observability/aws-otel-collector/pull/74 to fix this issue. Please try again once the PR is merged. Thank you for reaching out.

**Comment by mxiamxia** [comment]
Close the issue. Please feel free to open any new issue if you have.
